### PR TITLE
Enable resolution of AI services by bean name

### PR DESCRIPTION
**Pull request by aldettinger**
- Closes: #1087

### CODE REVIEW

**Review by geoand — APPROVED** [review]
LGTM, thanks!

File: core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java [hunk 1]
```diff
@@ -381,6 +382,12 @@ public void findDeclarativeServices(CombinedIndexBuildItem indexBuildItem,
 
             String imageModelName = chatModelName; // TODO: should we have a separate setting for this?
 
+            AnnotationInstance namedAnno = declarativeAiServiceClassInfo.annotation(NAMED);
+            Optional<String> beanName = Optional.empty();
```
Comment (cescoffier) [inline core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:386]: Is this the @Named annotation? We had an issue with the nonbinding aspect of that annotation in the past.

Comment (geoand) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:386]: It is yeah

Comment (aldettinger) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:386]: Yes, this is **@ Named** annotation. So the non binding aspect is that this annotation can't be implemented well in Quarkus ? Does it apply in this specific case ? Or maybe, it's just not recommended to use it ?

Comment (geoand) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:386]: It shouldn't be a problem in this case as it is entirely opt-in

**Pull request merged**